vignettes/plotting.Rmd

```diff
@@ -0,0 +1,128 @@
+```
+
+The Crunch web app provides a number of excellent visualizations and summary statistics to help communicate survey results to clients, but by its nature a web interface can't accommodate every type of plot or analysis. As a result one of our core design principles at Crunch is to let users access their data in as convenient a way as possible. This lets you seamlessly integrate Crunch data into your tool of choice, and customize how you communicate that data. This vignette goes through how you can use `ggplot2` and `tidyverse` tools to analyze and display crunch data using tidy tools. First we need to load a dataset into R. If this is a new process to you, see the [crunch vignettes](https://crunch.io/r/crunch/articles/crunch.html).
+
+```{r setup, include = FALSE}
+library(crunch)
+library(crplyr)
+ds <- loadDataset("SO_survey")
+```
+
+# ggplots
+
+You can now plot crunch variables using `ggplot2`'s `autoplot()` method. This will generate a plot which looks a lot like the plots produced on the app, but can be customized with ggplot methods.
+
+```{r basic card plots, fig.width=8, fig.height=6}
+library(ggplot2)
+autoplot(ds$CompanySize)
+```
+
+Crunch autoplot methods can produce three families of charts: dot plots, bar plots, and grid plots. These methods will automatically adjust for the dimensionality of the data which is sent into the plotting method. For instance, categorical array variables have two dimensions, so when they are plotted, you get a two dimensional plot.
+
+```{r CA plotting,  fig.width=8, fig.height=6}
+autoplot(ds$ImportantHiring)
+autoplot(ds$ImportantHiring, "grid")
+autoplot(ds$ImportantHiring, "bar")
+```
+
+Since the autoplot methods produce `ggplot` objects, you can customize and extend them just as you would any other ggplot. For example, you can change the theme of the plot or alter the color palette.
+
+```{r themes, fig.height=6, fig.width=8, warning=FALSE}
+p <- autoplot(ds$CompanySize)
+p + theme_grey()
+p +
+    geom_point(color = "red", size = 3) +
+    scale_x_log10()
+autoplot(ds$ImportantHiring) +
+    scale_color_brewer()
+```
+
+
+You can also define your own themes or use themes from other packages. This is helpful if you would like to send plots to a client using their style guide. For examples of sample corporate themes, take a look at the [ggtech](https://github.com/ricardo-bion/ggtech) package from AirBnb, or the [ggthemes](https://cran.r-project.org/web/packages/ggthemes/vignettes/ggthemes.html) package.
+
+## Extending ggplots
+
+`ggplot2` has a rich ecosystem of extensions which you can use to add functionality to Crunch plots. For example you can use [patchwork](https://github.com/thomasp85/patchwork) to combine several plots together, or add interactivity to your plots with [plotly](https://plot.ly/r/getting-started/). For a full list of what you can do with `ggplot` objects, see the [ggplot extensions](https://www.ggplot2-exts.org/) website.
+
+# Working with CrunchCubes
```

Comment:
I wonder whether we want to play up the CrunchCube concept. It might not be necessary to name it.